core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -170,6 +170,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        result.map{case (topicPartition, errorCode) => {
```

Comment:
This is probably fine, but I am wondering if this is needed. It seems the only case the broker will receive a StopReplicaRequest when it is still a leader is during topic deletion, which should not be allowed for the GroupMetadataTopic. Is there any case that the broker may receive the stop replica request when it is still the leader for the group metadata topic?